lib/init.php

```diff
@@ -57,3 +57,122 @@ function gutenberg_menu() {
 	);
 }
 add_action( 'admin_menu', 'gutenberg_menu', 9 );
+
+if ( ! function_exists( 'html_contains_block' ) ) {
+	/**
+	 * Recursively find a block by attribute.
```

Comment:
Algorithmically this is no longer recursive, nor is it conceptually. I'm not sure this is helpful to mention "recursive" because it might lead someone astray trying to figure out _how_ it's recursive.

We might say something more to the point about the goal:

> Returns whether the given HTML contains a block of the given type
> and, if provided, a given attribute and attribute value.

Reply:
Ah copy pasta sorry!